stdlib/logging/__init__.pyi

```diff
@@ -66,10 +66,20 @@ if sys.version_info >= (3, 12):
 _SysExcInfoType: TypeAlias = tuple[type[BaseException], BaseException, TracebackType | None] | tuple[None, None, None]
 _ExcInfoType: TypeAlias = None | bool | _SysExcInfoType | BaseException
 _ArgsType: TypeAlias = tuple[object, ...] | Mapping[str, object]
-_FilterType: TypeAlias = Filter | Callable[[LogRecord], bool]
 _Level: TypeAlias = int | str
 _FormatStyle: TypeAlias = Literal["%", "{", "$"]
 
+if sys.version_info >= (3, 12):
+    class _SupportsFilter(Protocol):
+        def filter(self, record: LogRecord) -> bool | LogRecord: ...
```

Comment:
Let's make the argument positional-only in the protocol, so if your class names it something else than `Record`, it will still match the protocol:

```suggestion
        def filter(self, __record: LogRecord) -> bool | LogRecord: ...
```

Reply:
I made the change in commit https://github.com/python/typeshed/pull/11018/commits/fe68d1aa8399e68ffc87c35834b997a42fdc5a73, but I’m still a little unclear: is the dunder `__record` a convention that the name is now “flexible”?

It doesn’t make the `record` arg (not capitalized `Record`) positional-only, though, that would [require a `/`](https://peps.python.org/pep-0570/) as well, right?

Reply:
Type-checkers special-case `__dunder`-named arguments and treat them as positional-only. This is a bit surprising, and we will soon be able to switch to the new `/` syntax.

Reply:
This is documented in PEP 484: https://peps.python.org/pep-0484/#positional-only-arguments

(PEP 484 predates [PEP 570](https://peps.python.org/pep-0570/), which was only implemented in Python 3.8 -- typeshed still supports Python 3.7.)